torchmultimodal/modules/encoders/clip_text_encoder.py

```diff
@@ -87,14 +102,21 @@ def initialize_parameters(self) -> None:
 
     def build_attention_mask(self) -> torch.Tensor:
         # To support torchscripting, we have to pass an int as fill_value
-        mask = torch.full((self.context_length, self.context_length), int(True)).triu(1)
-        return mask.to(device=None, dtype=torch.bool)
+        mask = torch.full(
+            (self.context_length, self.context_length), float("-inf")
```

Comment:
Why "-inf" here? Originally the mask was filled with "True" (or integer 1 for torchscript):

```
def build_attention_mask(self):
        mask = torch.full((self.context_length, self.context_length), True).triu(1)
        return mask.to(device=None, dtype=torch.bool)
```

cc @ebsmothers @sophiazhi

Reply:
Previously TorchText's transformer encoder was calling into their own MHA [here](https://github.com/pytorch/text/blob/ed44dc88539b51785df03a276410cf71adfe6bfd/torchtext/models/roberta/modules.py#L135-L137). But with the change to transformer encoder they are no longer doing this. So I think we need to make this change regardless